src/main/java/org/neo4j/shell/tools/imp/ExportKryoApp.java

```diff
@@ -0,0 +1,72 @@
+package org.neo4j.shell.tools.imp;
+
+import org.neo4j.cypher.export.CypherResultSubGraph;
+import org.neo4j.cypher.export.DatabaseSubGraph;
+import org.neo4j.cypher.export.SubGraph;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.GraphDatabaseAPI;
+import org.neo4j.shell.*;
+import org.neo4j.shell.impl.AbstractApp;
+import org.neo4j.shell.kernel.GraphDatabaseShellServer;
+import org.neo4j.shell.tools.imp.format.kryo.KryoWriter;
+import org.neo4j.shell.tools.imp.util.Config;
+import org.neo4j.shell.tools.imp.util.ProgressReporter;
+
+import java.io.FileOutputStream;
+
+/**
+ * Created by efulton on 5/5/16.
+ */
+public class ExportKryoApp extends AbstractApp {
+
+    {
+        addOptionDefinition( "o", new OptionDefinition( OptionValueType.MUST, "Output Binary file" ) );
+    }
+
+    @Override
+    public String getName() {
+        return "export-binary";
+    }
+
+    @Override
+    public GraphDatabaseShellServer getServer() {
+        return (GraphDatabaseShellServer) super.getServer();
+    }
+
+    @Override
+    public Continuation execute(AppCommandParser parser, Session session, Output out) throws Exception {
+        Config config = Config.fromOptions(parser);
+
+        String fileName = parser.option("o", null);
+
+        com.esotericsoftware.kryo.io.Output output = new com.esotericsoftware.kryo.io.Output(new FileOutputStream(fileName));
```

Comment:
Zipping makes a lot of sense.  The only reason I didn't use Inflator/Deflator input/output is it obscures the progress reporting since progress reports are based on size.  I think zipping it after the fact might make most sense since it'll take care of efficient compression for you.  I'm totally open to suggestion there though.  When I tried inflator/deflator on my database 300MB became 100MB.